### PR TITLE
[inductor][cpu] Fix the bias device check in _is_packable_linear

### PR DESCRIPTION
**Pull request by gavinwang269**
## Motivation

`_is_packable_linear` is the `extra_check` that gates CPU weight packing for `aten.mm`/`aten.addmm` in the freezing graph patterns. In the `addmm` branch the bias' device is checked against the wrong variable:

```python
bias_meta_value = linear_node.args[0].meta.get("val")
if (
    bias_meta_value is None
    or meta_value.device.type != "cpu"   # <-- meta_value is the loop variable
    ...
```

`meta_value` is left over from the loop right above over `[input_meta_value, weight_meta_value]`, so at this point *is* the weight fake whose device has just been proven to be `"cpu"`. The condition is never true and the bias device is effectively never validated.

## Changes

One-line fix: check `bias_meta_value.device.type` instead of `meta_value.device.type`.

## Behavior

This is a correctness fix to the predicate (unreachable-path hygiene), not a user-visible bug fix: the defect is real, but under the default configuration a mixed-device `aten.addmm` cannot be built under `FakeTensorMode` (mixed-device tensor args are rejected outside the allow-lists in `torch/_subclasses/fake_tensor.py`), so no graph that tracing/export can produce reaches the broken condition. For every graph the gate is reachable with, the predicate returns exactly what it returned before.

With `torch._functorch.config.fake_tensor_prefer_device_type` set to the bias device, a mixed-device addmm does construct; there the fixed check flips the gate from a wrongly-passing bias to the standard "extra check declined" fallback (return `False`, never raise), the same failure mode as every other check in this function, and the fix adds no new failure mode (`bias_meta_value` is already dereferenced by the neighbouring `.dim()`/`.size(0)` conditions).

Provable no-op for in-tree backends: XPU never reaches the bias branch (the preceding loop requires input/weight to be on CPU), and CUDA/MPS/MTIA never run this module's passes at all (callers gate on `torch._C._has_mkldnn` and `SUPPORTED_MKLDNN_DEVICES`). No new import-time work; the change is one expression inside an existing runtime predicate.

## Test Results

| env | collect | result |
|---|---|---|
| torch-2.15-cpu | test_mkldnn_pattern_matcher.py | 44 collected, Passed unchanged from base |
| torch-2.15-cpu | import torch._inductor.fx_passes.mkldnn_fusion | Passed (clean import) |
| lintrunner (PYFMT,RUFF) | mkldnn_fusion.py | Passed (0 issues) |

No test added deliberately: reaching the repaired condition requires a graph that `FakeTensorMode` refuses to build by default, so a test would have to hand-construct the graph and mock the matcher, asserting a shape no real pipeline produces (an earlier version of this PR shipped such a test; the review correctly identified that it costs more than it protects, and it has been dropped).

Verified locally:

```console
cd /tmp && unset PYTHONPATH
python -c "import torch._inductor.fx_passes.mkldnn_fusion"   # clean import
python -m pytest $PT/test/inductor/test_mkldnn_pattern_matcher.py -q --co
# 44 tests collected, unchanged from the base commit
lintrunner --take PYFMT,RUFF torch/_inductor/fx_passes/mkldnn_fusion.py
# ok No lint issues
```

## Notes

`_is_packable_convolution` checks only input and weight (against `SUPPORTED_MKLDNN_DEVICES`) and never its bias, so the linear and conv predicates do not agree. Aligning them can only make the conv gate stricter, which is a behavior change worth its own discussion rather than a rider on this fix; happy to add it here or send it separately on a maintainer's word.

Keeping `!= "cpu"` (rather than `SUPPORTED_MKLDNN_DEVICES`) in the fixed condition is deliberate: the preceding input/weight loop already requires `cpu`, and `XpuMkldnnDeviceOp` inherits `pack_linear`/`pack_linear_weight`, which raise `NotImplementedError` in `MkldnnDeviceOpBase`.

The other `cpu`/`xpu` literals in this file (`_get_mkldnn_device_op`'s dispatch, the grouped-gemm and RNN gates, the transposed-conv XPU exclusion) are deliberately left as they are: they encode what oneDNN actually supports (`mkldnn::_*` ops exist only on CPU/XPU), and the `RuntimeError` in `_get_mkldnn_device_op` for other devices is an explicit unsupported-backend signal. Turning that dispatch into a registration table would be a separate abstraction change, and out-of-tree fusion passes already have `post_grad_custom_pre_pass` / `register_backend_for_device` today.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by gavinwang269** [comment]
Hi @jansel, this PR has been updated per the Claude review: the test has been dropped (the review correctly identified it cost more than it protected), and the Behavior section now states up front that this is unreachable-path hygiene. The one-line fix itself is unchanged. CI is green on the current head. Could you take another look when you get a chance?

**Comment by jansel** [comment]
@claude review again, do we need to add a test here?

**Review by jansel — APPROVED** [review]

**Comment by gavinwang269** [comment]
Hi @jgong5, could you take a look at this PR when you have a chance? It's a small fix (1 line changed) to the bias device check in `_is_packable_linear` in `mkldnn_fusion.py`, and it falls under the CPU inductor merge rules where your approval is required.

Current status:
- Reviewed and approved by @jansel (plus a follow-up Claude re-review confirming the trimmed test approach)
- All CI green; the only failing check is `torchtitan_models_integration`, which fails with `No module named 'fla'` — a CI environment dependency issue unrelated to this change (same job passes on main)

cc @CaoE in case you're more familiar with this area.

Thanks!
